jaxlib/handle_pool.h

```diff
@@ -0,0 +1,107 @@
+/* Copyright 2019 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef JAXLIB_HANDLE_POOL_H_
+#define JAXLIB_HANDLE_POOL_H_
+
+#include <vector>
+
+#include "absl/base/thread_annotations.h"
+#include "absl/synchronization/mutex.h"
+
+namespace jax {
+namespace {
+
+
+// To avoid creating cublas/cusolver contexts in the middle of execution, we maintain
+// a pool of them.
+template <typename handle_t, typename stream_t>
```

Comment:
Stylistically, I'd probably call this `Handle` and `Stream` or `HandleType` and `StreamType`.

(We mostly follow Google C++ style, in which types start with uppercase letters: https://google.github.io/styleguide/cppguide.html#Type_Names)

Reply:
done. 700ac820e2b76b47a09250cc140392a8ebb8564b